realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/CredentialsTests.kt

```diff
@@ -82,22 +101,28 @@ class CredentialsTests {
         assertFailsWith<IllegalArgumentException> { Credentials.apple(TestHelper.getNull()) }
     }
 
-    @Ignore("FIXME: Awaiting ObjectStore support")
     @Test
     fun customFunction() {
-        TODO()
+        val mail = "myfakemail@mongodb.com"
+        val id = 666
+        val creds = mapOf(
+                "mail" to "myfakemail@mongodb.com",
+                "id" to 666
+        ).let { Credentials.customFunction(Document(it)) }
+        assertEquals(Credentials.IdentityProvider.CUSTOM_FUNCTION.id, creds.identityProvider.id)
+        assertTrue(creds.asJson().contains(mail))
+        assertTrue(creds.asJson().contains(id.toString()))
     }
 
-    @Ignore("FIXME: Awaiting ObjectStore support")
     @Test
     fun customFunction_invalidInput() {
-        TODO()
+        assertFailsWith<IllegalArgumentException> { Credentials.customFunction(null) }
     }
 
     @Test
     fun emailPassword() {
         val creds = Credentials.emailPassword("foo@bar.com", "secret")
-        assertEquals("local-userpass", creds.identityProvider.id)
+        assertEquals(Credentials.IdentityProvider.EMAIL_PASSWORD.id, creds.identityProvider.id)
```

Comment:
While this is much better. Is there a reason we need the `id`? Couldn't we just compare directly against the `identityProvider` since it is an enum. The same is true in a lot of other places